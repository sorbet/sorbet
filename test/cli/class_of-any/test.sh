#!/usr/bin/env bash

tmp="$(mktemp -d)"
infile="test/cli/class_of-any/class_of-any.rb"
cp "$infile" "$tmp"

cwd="$(pwd)"
cd "$tmp" || exit 1

if "$cwd/main/sorbet" --silence-dev-message -a class_of-any.rb 2>&1; then
  echo "Expected to fail!"
  exit 1
fi

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make that the autocorrect applied
cat class_of-any.rb

rm class_of-any.rb
rm "$tmp"
