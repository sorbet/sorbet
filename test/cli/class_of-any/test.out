class_of-any.rb:9: `T.class_of` must wrap each individual class type, not the outer `T.any` https://srb.help/5004
     9 |sig { returns(T.class_of(T.any(A, T.any(B, C)))) }
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Done
    class_of-any.rb:9: Replaced with `T.any(T.class_of(B), T.class_of(C), T.class_of(A))`
     9 |sig { returns(T.class_of(T.any(A, T.any(B, C)))) }
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

class_of-any.rb:12: `T.class_of` needs a class or module as its argument https://srb.help/5004
    12 |sig { returns(T.class_of(T.any(A, D))) }
                      ^^^^^^^^^^^^^^^^^^^^^^^
Errors: 2

--------------------------------------------------------------------------

# typed: true
extend T::Sig

class A; end
class B; end
class C; end
class D < A; end

sig { returns(T.any(T.class_of(B), T.class_of(C), T.class_of(A))) }
def foo; end

sig { returns(T.class_of(T.any(A, D))) }
def bar; end
